src/utils/generator.js

```diff
@@ -136,9 +137,20 @@ class TemporalBuildNodeEntry extends GeneratorEntry {
   buildNode: void | GeneratorBuildNodeFunction;
   dependencies: void | Array<Generator>;
   isPure: void | boolean;
+  mutatesOnly: void | Array<Value>;
```

Comment:
In the constructor, can you add invariant(s) that ensure that `isPure` and `mutatesOnly` are mutually exclusive? Also, an invariant that the values listed in `mutatesOnly` are a subset of the `args`.